packages/core/src/app/payment/PaymentSubmitButton.test.tsx

```diff
@@ -26,158 +26,155 @@ describe('PaymentSubmitButton', () => {
         PaymentSubmitButtonTest = (props) => (
             <CheckoutProvider checkoutService={checkoutService}>
                 <LocaleContext.Provider value={localeContext}>
-                    <PaymentSubmitButton {...props} />
+                    <PaymentSubmitButton
+                        isPaymentDataRequired
+                        {...props}
+                    />
                 </LocaleContext.Provider>
             </CheckoutProvider>
         );
     });
 
-    it('matches snapshot with rendered output', () => {
-        const component = render(<PaymentSubmitButtonTest />);
-
-        expect(component).toMatchSnapshot();
-    });
-
-    it('forwards props to button', () => {
-        const component = mount(<PaymentSubmitButtonTest isDisabled />);
+    it('renders button with default label', () => {
+        render(<PaymentSubmitButtonTest />);
 
-        expect(component.find(Button).props()).toEqual(
-            expect.objectContaining({
-                disabled: true,
-            }),
+        expect(screen.getByTestId(submitButtonId)).toHaveTextContent(
+            languageService.translate('payment.place_order_action'),
         );
```

Comment:
RTL encourages you to assert like this as stated here, https://testing-library.com/docs/queries/about/#priority.
Therefore, `toHaveTextContent` can be avoided.
```
expect(screen.getByText(languageService.translate('payment.place_order_action'))).toBeInTheDocument();
```

Reply:
Oh, ok. Thanks! Give me a sec I will update the PR